File: src/main/java/org/taktik/connector/business/mycarenet/attestv3/service/AttestService.java
package org.taktik.connector.business.mycarenet.attestv3.service;

import org.taktik.connector.technical.exception.TechnicalConnectorException;
import org.taktik.connector.technical.service.sts.security.SAMLToken;
import be.fgov.ehealth.mycarenet.attest.protocol.v3.CancelAttestationResponse;
import be.fgov.ehealth.mycarenet.attest.protocol.v3.SendAttestationResponse;
import be.fgov.ehealth.mycarenet.commons.protocol.v3.SendRequestType;

public interface AttestService {
   SendAttestationResponse sendAttestion(SAMLToken var1, SendRequestType var2) throws TechnicalConnectorException;

   CancelAttestationResponse cancelAttestion(SAMLToken var1, SendRequestType var2) throws TechnicalConnectorException;
}

File: src/main/kotlin/org/taktik/connector/business/agreement/service/AttestService.kt
package org.taktik.connector.business.agreement.service

import be.fgov.ehealth.mycarenet.attest.protocol.v3.CancelAttestationResponse
import be.fgov.ehealth.mycarenet.attest.protocol.v3.SendAttestationResponse
import be.fgov.ehealth.mycarenet.commons.protocol.v4.SendRequestType
import org.taktik.connector.technical.exception.TechnicalConnectorException
import org.taktik.connector.technical.service.sts.security.SAMLToken

interface AttestService {
    @Throws(TechnicalConnectorException::class)
    fun sendAttestion(token: SAMLToken?, request: SendRequestType?): SendAttestationResponse?

    @Throws(TechnicalConnectorException::class)
    fun cancelAttestion(token: SAMLToken?, request: SendRequestType?): CancelAttestationResponse?
}
